lib/radar_client.js

```diff
@@ -454,35 +433,44 @@ Client.prototype._restore = function() {
   }
 };
 
-Client.prototype._sendMessage = function(message) {
-  var memorized = this._memorize(message);
-  this.emit('message:out', message);
+Client.prototype._sendMessage = function(request) {
+  var memorized = this._memorize(request),
+      ack = request.getAttr('ack');
+
+  this.emit('message:out', request.getMessage());
 
   if (this._socket && this.manager.is('activated')) {
-    this._socket.sendPacket('message', JSON.stringify(message));
+    this._socket.sendPacket('message', JSON.stringify(request.getMessage()));
   } else if (this._isConfigured) {
     this._restoreRequired = true;
-    if (!memorized || message.ack) {
-      this._queuedMessages.push(message);
+    if (!memorized || ack) {
+      this._queuedMessages.push(request);
     }
     this.manager.connectWhenAble();
   }
 };
 
 Client.prototype._messageReceived = function (msg) {
-  var message = JSON.parse(msg);
+  var message = JSON.parse(msg),
+      op = Response.getAttr(message, 'op'),
+      to = Response.getAttr(message, 'to');
+
   this.emit('message:in', message);
-  switch (message.op) {
+
+  switch (op) {
     case 'err':
     case 'ack':
     case 'get':
-      this.emitNext(message.op, message);
+      var response = new Response(message);
+      this.emitNext(op, new Response(message));
       break;
+
     case 'sync':
-      this._batch(message);
+      this._batch(new Response(message)); 
       break;
+
     default:
-      this.emitNext(message.to, message);
+      this.emitNext(to, message);
```

Comment:
Now that I have removed the use of **throw**, this will work.  It also requires that I change the first check in **response.validate** so that it does not log 'missing op' when it sees a response message like this: **{ open: 1 }**